junction/src/main/java/org/killbill/billing/junction/plumbing/billing/DefaultInternalBillingApi.java

```diff
@@ -151,15 +154,22 @@ private void addBillingEventsForBundles(final List<SubscriptionBaseBundle> bundl
 
         }
 
+        final List<Tag> bundleTags = getTagsForObjectType(ObjectType.BUNDLE, tags);
+        final Map<UUID, List<SubscriptionBase>> subscriptionsForAccount = subscriptionApi.getSubscriptionsForAccount(context);
+
         for (final SubscriptionBaseBundle bundle : bundles) {
             final DryRunArguments dryRunArgumentsForBundle = (dryRunArguments != null &&
                                                               dryRunArguments.getBundleId() != null &&
                                                               dryRunArguments.getBundleId().equals(bundle.getId())) ?
                                                              dryRunArguments : null;
-            final List<SubscriptionBase> subscriptions = subscriptionApi.getSubscriptionsForBundle(bundle.getId(), dryRunArgumentsForBundle, context);
+            final List<SubscriptionBase> subscriptions;
+            if (dryRunArgumentsForBundle == null || dryRunArgumentsForBundle.getAction() == null) {
+                subscriptions = subscriptionsForAccount.get(bundle.getId());
+            } else {
+                subscriptions = subscriptionApi.getSubscriptionsForBundle(bundle.getId(), dryRunArgumentsForBundle, context);
```

Comment:
Yup, I'm happy not optimizing that path, since it's not the common one. But let's add a comment to explain that we deliberately ignored the problem here, so we remember.